sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -313,7 +314,7 @@ object InMemoryFileIndex extends Logging {
         // to retrieve the file status with the file block location. The reason to still fallback
         // to listStatus is because the default implementation would potentially throw a
         // FileNotFoundException which is better handled by doing the lookups manually below.
-        case _: DistributedFileSystem if !ignoreLocality =>
+        case (_: DistributedFileSystem | _: ViewFileSystem) if !ignoreLocality =>
```

Comment:
Hi, @manuzhang .
Which test did you referring in the PR description? Could you add a UT explicitly?
```
## How was this patch tested?

Existing tests.
```

Reply:
Sorry, I didn't find a UT for `DistributedFileSystem`. Could you suggest on how to test it ?

Reply:
`Mock`?
```
$ git grep 'DistributedFileSystem' | grep Suite
core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala:import org.apache.hadoop.hdfs.{DFSInputStream, DistributedFileSystem}
core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala:    val dfs = mock(classOf[DistributedFileSystem])
sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala:      "fs.file.impl" -> classOf[MockDistributedFileSystem].getName,
sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala:// This file system is for SPARK-14959 (DistributedFileSystem will throw an exception
sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala:class MockDistributedFileSystem extends RawLocalFileSystem {
```

Reply:
BTW, without a test case, your contribution is not protected from accidental removal. That's the reason why we ask a UT always.

Reply:
thanks, UT added